From akpm@osdl.org Tue Aug 22 07:47:01 2006
Message-Id: <200608221429.k7METABH016496@shell0.pdx.osdl.net>
Subject: PCI: i386 mmconfig: don't forget bus number when setting fallback_slots bits
From: Daniel Ritz <daniel.ritz-ml@swissonline.ch>
To: greg@kroah.com
Cc: akpm@osdl.org, daniel.ritz-ml@swissonline.ch, daniel.ritz@gmx.ch
Date: Tue, 22 Aug 2006 07:29:09 -0700

From: Daniel Ritz <daniel.ritz-ml@swissonline.ch>

On i386 PCI mmconfig forgets the bus number when setting the fallback_slots
bits which means fallback to conf1 only works for bus 0.

Signed-off-by: Daniel Ritz <daniel.ritz@gmx.ch>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/i386/pci/mmconfig.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/arch/i386/pci/mmconfig.c
+++ gregkh-2.6/arch/i386/pci/mmconfig.c
@@ -178,7 +178,7 @@ static __init void unreachable_devices(v
 				pci_exp_set_dev_base(addr, k, PCI_DEVFN(i, 0));
 			if (addr == 0 ||
 			    readl((u32 __iomem *)mmcfg_virt_addr) != val1) {
-				set_bit(i, fallback_slots);
+				set_bit(i + 32*k, fallback_slots);
 				printk(KERN_NOTICE
 			"PCI: No mmconfig possible on %x:%x\n", k, i);
 			}
